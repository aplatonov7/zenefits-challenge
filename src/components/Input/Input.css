.Input__container {
  padding: 0.75rem;
  transition-property: background;
  transition-duration: 0.2s;
  transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
}

.Input__container.colored {
  background-color: #4285F4;
}

.Input__text {
  width: 100%;
  border: none;
  height: 3rem;
  padding: 1rem 3rem 1rem 1rem;
  font-size: 1.15rem;
  box-shadow: 0 1px 4px rgba(0, 0, 0,   0.3);
}

.Input__clear-button {
  position: absolute;
  top: 1.25rem;
  right: 1.25rem;
  font-size: 0;
  cursor: pointer;
  padding: 0.25rem 0.5rem;
}

.Input__clear-button:after {
  content: 'x';
  font-size: 1.5rem;
  line-height: 1;
  color: #777;
}

/* Styles for google maps searchBox */

.pac-item {
  padding: 0.5rem 0.75rem;
}

.pac-item:last {
  padding-bottom: 0;
}