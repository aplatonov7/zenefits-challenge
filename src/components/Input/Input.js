import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import './Input.css' 

const Input = ({ displayWrapper, setRef, onClear }) => (
  <div className={cx("Input__container", {
    colored: displayWrapper
  })}>
    <input 
      type="text" 
      className="Input__text" 
      placeholder="Search places"
      aria-label="Search places"
      ref={setRef} 
    />
    <button className="Input__clear-button" onClick={onClear}>Clear Input</button>
  </div>
)

Input.propTypes = {
  displayWrapper: PropTypes.bool,
  onClear: PropTypes.func,
  setRef: PropTypes.func,
}

export default Input