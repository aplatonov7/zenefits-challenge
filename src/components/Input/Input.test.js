import React from 'react'

import Input from './Input'

describe('Input component', () => {
  let component

  beforeEach(() => {
    component = shallow(<Input />)
  })

  it('Should match snapshot', () => {
    const props = { diplayWrapper: true, onClear: () => {} }
    expect(renderer.create(<Input {...props} />).toJSON()).toMatchSnapshot()
  })

  it('Should display colored wrapper when diplayWrapper prop is set', () => {
    component.setProps({ displayWrapper: true })
    expect(component.hasClass('colored')).toBe(true)
  })

  it('Should fire onClear prop on clear button click', () => {
    const mockFn = jest.fn()
    component.setProps({ onClear: mockFn })
    component.find('button.Input__clear-button').simulate('click')
    expect(mockFn).toBeCalled()
  })
})