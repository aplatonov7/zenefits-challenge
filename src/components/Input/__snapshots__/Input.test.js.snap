// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should match snapshot 1`] = `
<div
  className="Input__container"
>
  <input
    aria-label="Search places"
    className="Input__text"
    placeholder="Search places"
    type="text"
  />
  <button
    className="Input__clear-button"
    onClick={[Function]}
  >
    Clear Input
  </button>
</div>
`;
